linkerd/core/src/main/scala/io/buoyant/linkerd/Announcer.scala

```diff
@@ -1,11 +1,30 @@
 package io.buoyant.linkerd
 
-import com.twitter.finagle.{Announcement, Path}
-import com.twitter.util.Future
+import com.twitter.finagle.{Announcement, Failure, Path}
+import com.twitter.util.{Closable, Future, Return, Throw}
+import java.util.concurrent.atomic.AtomicReference
 import java.net.InetSocketAddress
 
 abstract class Announcer {
-  val scheme: String
+  def scheme: String
+  def apply(addr: InetSocketAddress, name: Path): Closable
```

Comment:
I actually prefer your second example over your first.  I think `apply` is overused in scala.  But that's just, like, my opinion, man.  FFTI.